src/modules/tools/temperaturecontrol/TemperatureControl.cpp

```diff
@@ -510,8 +519,54 @@ void TemperatureControl::pid_process(float temperature)
 
 void TemperatureControl::on_second_tick(void *argument)
 {
+
+    // If waiting for a temperature to be reach, display it to keep host programs up to date on the progress
     if (waiting)
         THEKERNEL->streams->printf("%s:%3.1f /%3.1f @%d\n", designator.c_str(), get_temperature(), ((target_temperature <= 0) ? 0.0 : target_temperature), o);
+
+    // Check whether or not there is a temperature runaway issue, if so stop everything and report it
+    if(THEKERNEL->is_halted()) return;
+    
+    if( this->target_temperature <= 0 ){ // If we are not trying to heat, state is NOT_HEATING
+        this->runaway_state = NOT_HEATING;
+        this->runaway_timer = 0;
+    }else{
+        switch( this->runaway_state ){
+            case NOT_HEATING: // If we were previously not trying to heat, but we are now, change to state WAITING_FOR_TEMP_TO_BE_REACHED
+                if( this->target_temperature > 0 ){
+                    this->runaway_state = WAITING_FOR_TEMP_TO_BE_REACHED;
+                    this->runaway_heating_timer = 0;
+                }
+                break;
+            case WAITING_FOR_TEMP_TO_BE_REACHED: // In we are in state 1 ( waiting for temperature to be reached ), and the temperature has been reached, change to state TARGET_TEMPERATURE_REACHED
+                if( this->get_temperature() >= this->target_temperature ){
+                    this->runaway_state = TARGET_TEMPERATURE_REACHED;
+                }
+                this->runaway_heating_timer++;
+                if( this->runaway_heating_timer > this->runaway_heating_timeout && this->runaway_heating_timeout != 0 ){
+                    this->runaway_heating_timer = 0;
```

Comment:
Could `this->runaway_heating_timer` and `this->runaway_timer` be merged to save 1 byte?
From a once-over,  it looks like they are never used at the same time.